pkg/reconciler/v1alpha1/eventlistener/eventlistener.go

```diff
@@ -298,14 +319,32 @@ func (r *Reconciler) reconcileDeployment(logger *zap.SugaredLogger, el *v1alpha1
 			Name:      "config-logging",
 			MountPath: "/etc/config-logging",
 		}},
-		Env: []corev1.EnvVar{{
-			Name: "SYSTEM_NAMESPACE",
-			ValueFrom: &corev1.EnvVarSource{
-				FieldRef: &corev1.ObjectFieldSelector{
-					FieldPath: "metadata.namespace",
-				},
-			},
-		}},
+		Env: env,
+	}
+	var (
+		tolerations                          []corev1.Toleration
+		nodeSelector, annotations, podlabels map[string]string
+		serviceAccountName                   string
+	)
+	podlabels = mergeMaps(el.Labels, GenerateResourceLabels(el.Name))
+
```

Comment:
same comment as with `serviceType` -> it seems like we could also do some of this in the webhook via EL's SetDefaults.